src/OrbitGl/ThreadBar.h

```diff
@@ -30,6 +32,7 @@ class ThreadBar : public CaptureViewElement, public std::enable_shared_from_this
   void SetCaptureData(CaptureData* capture_data) { capture_data_ = capture_data; }
 
  protected:
+  TimeGraphLayout* layout_;
```

Comment:
You could move this to `CaptureViewElement`, I think it's fundamental enough to be in the base class. And think about adding a CHECK for non-null during construction since most Draw / UpdatePrimitives methods rely on this being non-null.

Reply:
I was 50-50, not sure sure how much fundamental it was... Still not sure about that, but will follow your advice and changing it :)